frictionless/portals/github/manager.py

```diff
@@ -1,45 +1,190 @@
-from typing import Optional
+from collections import namedtuple
+from typing import Any, Dict, List, NamedTuple, Union
+from github.ContentFile import ContentFile
+from github.Repository import Repository
 from .control import GithubControl
-from ...platform import platform
-from ...resource import Resource
+from ...exception import FrictionlessException
+from ...catalog import Catalog
 from ...package import Package
 from ...package import Manager
+from ...platform import platform
+from ...resource import Resource
 
 
 class GithubManager(Manager[GithubControl]):
+    """Read and write data from/to Github"""
 
     # Read
 
-    # TODO: implement
-    def read_catalog(self):
-        pass
-
-    # TODO: improve
-    def read_package(self, *, user: Optional[str] = None, repo: Optional[str] = None):
-        client = platform.github.Github()
-        user = user or self.control.user
-        repo = repo or self.control.repo
-        assert user
-        assert repo
-        location = "/".join([user, repo])
-        repository = client.get_repo(location)
-        branch = repository.default_branch
+    def read_catalog(self) -> Catalog:
+        packages = []
+        query: Dict[str, Any] = {}
+
+        if self.control.repo or self.control.user or self.control.search:
```

Comment:
I would suggest raising an exception here like:

```
if not self.control.repo and not ...:
    raise
```

It will allow to write the following code without indentation